File: sbt-test/basic/out-of-order-reference/libraryA/src/main/java/test/LibraryA.java
package test;

import java.nio.ByteBuffer;
import java.nio.charset.Charset;

public class LibraryA {
    public static String wasteTime(String input) {
        Charset utf8 = Charset.forName("UTF-8");
        ByteBuffer buffer = utf8.encode(input);
        return utf8.decode(buffer).toString();
    }
}

File: sbt-test/scala/out-of-order-reference/libraryA/src/main/scala/test/LibraryA.scala
package test

import java.nio.charset.Charset

object LibraryA {
  def wasteTime(input: String): String  = {
    val utf8 = Charset.forName("UTF-8")
    val buffer = utf8.encode(input)
    utf8.decode(buffer).toString
  }
}
